File: 4_cnn_fb40/2_mxnet_ce/mpi_run.pl
#!/usr/bin/perl
use strict;
use warnings;
$| = 1; #### flush STDOUT

my $ngpu = $ARGV[0];

system("rm -rf stop");
my $count = 0;
while (!-e "stop")
{
    # system("perl train_init.pl -1");
    system("perl train.pl $ngpu >train.$count.log 2>&1");
    sleep(60);
    $count++;
    print "MPI run died ,restart $count\n";
}

File: 4_cnn_fb40/2_mxnet_ce/mpi_run.py
#!/opt/tool/anaconda3.5/bin/python3
#coding=utf-8

import os,sys
import time

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print('Usage: py ngpu')
    ngpu = int(sys.argv[1])
    
    if os.path.exists('stop'):
        os.remove('stop')
    
    count = 0
    while not os.path.exists('stop'):
        # cmdline = 'perl train_init.pl -1'
        cmdline = 'perl train.pl {} >train.{}.log 2>&1'.format(ngpu, count)
        os.system(cmdline)
        time.sleep(60)
        count += 1
        print('MPI run died, restart ', count)
